### PR TITLE
Add AddBillingSetupExample

### PR DESCRIPTION
**Pull request by devchas**

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+import com.google.protobuf.StringValue;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Creates a billing setup for a customer. A billing setup is a link between a payments account and
+ * a customer. The new billing setup can either reuse an existing payments account, or create a new
+ * payments account with a given payments profile. Billing setups are applicable for clients on
+ * monthly invoicing only. See here for details about applying for monthly invoicing:
+ * https://support.google.com/google-ads/answer/2375377 In the case of consolidated billing, a
```
Comment (AndrewMBurke) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:41]: Please add a new line or punctuation after the URL for readability.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param billingSetup The instance of BillingSetup whose starting date time will be set.
+   * @return Billing Setup with updated start and end date time.
+   */
+  private BillingSetup setBillingSetupDateTimes(
+      GoogleAdsClient googleAdsClient, long customerId, BillingSetup billingSetup)
+      throws Exception {
+    // The query to search existing approved billing setups in the end date time descending
+    // order.
+    // See GetBillingSetup.cs for a more detailed example of requesting billing setup
```
Comment (AndrewMBurke) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:209]: GetBillingSetup.java

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+        // Sets the new billing setup to start one day after the ending date time.
+        billingSetupBuilder.setStartDateTime(
+            StringValue.of(lastEndingDateTime.plusDays(1).toString("yyyy-MM-dd")));
+
+        // Sets the new billing setup to end one day after the starting date time.
+        billingSetupBuilder.setEndDateTime(
+            StringValue.of(lastEndingDateTime.plusDays(2).toString("yyyy-MM-dd")));
+      } else {
+        // Otherwise, the only acceptable start time is TimeType.Now.
```
Comment (AndrewMBurke) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:253]: For clarity, perhaps "the only acceptable start time is DateTime.now()"?  TimeType.Now is the C# value.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID containing the BillingSetup to remove.
+   * @param billingSetup The instance of BillingSetup whose starting date time will be set.
+   * @return Billing Setup with updated start and end date time.
+   */
+  private BillingSetup setBillingSetupDateTimes(
+      GoogleAdsClient googleAdsClient, long customerId, BillingSetup billingSetup)
+      throws Exception {
+    // The query to search existing approved billing setups in the end date time descending
```
Comment (nwbirnie) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:207]: Could rephrase this to "Defines a query to ...."

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+          googleAdsServiceClient.search(Long.toString(customerId), query);
+
+      BillingSetup.Builder billingSetupBuilder = billingSetup.toBuilder();
+      if (searchPagedResponse.getPage().getResponse().getResultsCount() > 0) {
+        // Retrieves the ending date time of the last billing setup.
+        StringValue lastEndingDateTimeString =
+            searchPagedResponse
+                .getPage()
+                .getResponse()
+                .getResults(0)
```
Comment (nwbirnie) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:231]: Would it be clearer to do searchPagedResponse.iterator().next()?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+  private BillingSetup createBillingSetup(
+      long customerId, String paymentsAccountId, String paymentsProfileId) throws Exception {
+    BillingSetup.Builder billingSetupBuilder = BillingSetup.newBuilder();
+
+    // Sets the appropriate payments account field.
+    if (paymentsAccountId != null) {
+      // If a payments account id has been provided, set PaymentsAccount to its resource
+      // name. You can list available payments accounts via the PaymentsAccountService's
+      // ListPaymentsAccounts method.
+      billingSetupBuilder.setPaymentsAccount(
```
Comment (nwbirnie) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:174]: Has this been tested?

Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:174]: I do not have access to an account with monthly invoicing, so I relied on the golden for this one. @AndrewMBurke can you verify this has been tested for DotNet?

@nwbirnie do you have a specific concern here?

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java (hunk 5 above)
Comment (devchas) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/billing/AddBillingSetup.java:231]: Iterator is not available on `searchPagedResponse`. You could do `searchPagedResponse.iterateAll().iterator().next()` but I think that's actually more confusing.
